src/dfx/src/commands/language_service.rs

```diff
@@ -40,7 +44,35 @@ pub fn exec(env: &dyn Environment, opts: LanguageServiceOpts) -> DfxResult {
             .get_defaults()
             .get_build()
             .get_packtool();
-        let package_arguments = package_arguments::load(env.get_cache().as_ref(), packtool)?;
+
+        let mut package_arguments = package_arguments::load(env.get_cache().as_ref(), packtool)?;
+
+        // Include actor alias flags
+        let canister_names = config
+            .get_config()
+            .get_canister_names_with_dependencies(None)?;
+        let agent_env = create_agent_environment(env, None /* opts.network */)?;
+        let canister_id_store = CanisterIdStore::for_env(&agent_env)?;
```

Comment:
We can use the Environment here, already in scope.
```suggestion
        let canister_id_store = CanisterIdStore::for_env(env)?;
```

Reply:
Using `env` instead of `agent_env` causes the following error at runtime:
`'internal error: entered unreachable code: NetworkDescriptor only available from an AgentEnvironment'`
Is there a way to avoid this using the provided (non-Agent)Environment? 

Thank you for taking a look at this PR.

Reply:
Oh yes, you are right.  I've submitted https://github.com/dfinity/sdk/pull/2376, which will provide `CanisterIdStore::new` for this purpose.